teps/0060-remote-resource-resolution.md

```diff
@@ -0,0 +1,370 @@
+
+Tekton Pipelines adds out-of-the-box support for referencing catalog tasks
+directly from the open source catalog by packaging a "catalog resolver" as
+part of the latest `release.yaml`.
+
+A team keeps all of their CI/CD workflows in a cloud bucket so they can
+quickly add new features. They want to be able to refer to the tasks and
+pipelines stored in that bucket rather than `kubectl apply` when a new
+workflow is added.
+
+A platform builder accepts workflows imported from other Tekton-compliant
+platforms. When a Pipeline is imported that includes a reference to a Task
+in a Remote that the platform doesn't already have it goes out and fetches
+that Task, importing it alongside the pipeline without the user having to
+explicitly provide it.
+
+A company requires their CI/CD Tasks to be audited to check that they adhere to
+specific compliance rules before the Tasks can be used in their app teams'
+Pipelines. The company hosts a private internal catalog and requires all the
+app teams to build their Pipelines using only that catalog. To make this easier
+all of the CI/CD clusters in the company are configured with a single remote
+resource resolver - a CatalogRef resolver - that points to the internal
+catalog. When a new Task is written it must go through an audit and review
+process before it is allowed to be merged into the internal catalog for all
+teams to use.
+
+## Requirements
+
+- Ensure that resources fetched from remote locations are given the same
+  treatment as those applied directly to the cluster - they should be
+  defaulted, mutated, and validated just as they are when `kubectl apply`ing
+  directly to a cluster. This doesn't necessarily mean that remote resources
+  will be passed through `kubectl apply`, only that the processing steps
+  involved are equivalent.
+- Provide a clear, well-documented interface between Pipelines' role as
+  executor of Tekton resources and Resolvers' role fetching those Tekton
+  resources before execution can begin.
+- Include a configurable timeout for async resolution so operators can tune the
+  allowable delay before a resource resolution is considered failed.
+- Avoid blocking reconciler threads while fetching a remote resource is in
+  progress.
+- The implementation should be compatibile with the existing flagged Tekton
+  Bundles support, with a view to replacing Tekton Pipeline's built-in
+  resolvers with the new solution.
+- Allow RBAC to be managed per remote source so that the permissions to fetch
+  and return Tekton resources to a reconciler are limited to only those needed
+  by each type of source.
+- Support credentials to access a remote both per-source or per-workload,
+  allowing operators to decide if they manage the creds to fetch Tekton
+  resources or require their users to provide them at runtime.
+- Support Tasks, Pipelines, Custom Tasks and any other resources that Tekton
+  Pipelines might choose to add support for (e.g. Steps?).
```

Comment:
It's not limited to pipeline though right, it could be similar for triggers objects too (binding, interceptors, …)

Reply:
OK thanks for clarifying.

Reply:
I'm trying to read between the lines a bit here @vdemeester and I want to know if I'm misunderstanding:

If I rewrite this proposal to remove mention of pipelines, and push it more in the direction of a library and/or independent Tekton component, would that alleviate some of your current concerns?